# Los Datos

Los conjuntos de datos que usaremos corresponden a la ejecución de ingresos públicos en Brasil durante los años 2013 al 2021. La **ejecución de ingresos públicos** se refiere al proceso mediante el cual un gobierno recauda y gestiona sus ingresos fiscales y no fiscales para financiar sus actividades y programas. En el contexto de Brasil, y en los datos que mencionas, la ejecución de ingresos públicos implica cómo se administraron los recursos que el gobierno planeaba recibir (ingresos previstos) y lo que efectivamente recaudó (ingresos realizados) durante un periodo específico, en este caso, desde 2013 hasta 2021. En Brasil, los ingresos públicos pueden venir de:

- **Impuestos:** Como el Imposto de Renda (IR) y el Imposto sobre Produtos Industrializados (IPI).

- **Contribuciones Sociales:** Como las contribuciones para la seguridad social (INSS).

- **Tasas y Tarifas:** Que pueden ser cobradas por servicios específicos.

- **Otros Ingresos:** Como multas, indemnizaciones, y la venta de activos.


Las columnas presentes en estos archivos son:

- `código órgão superior`: Código numérico que identifica la entidad gubernamental superior.

- `nome órgão superior`: Nombre de la entidad gubernamental superior.

- `código órgão`: Código numérico que identifica la entidad gubernamental específica.

- `nome órgão`: Nombre de la entidad gubernamental específica.

- `código unidade gestora`: Código numérico de la unidad gestora responsable.

- `nome unidade gestora`: Nombre de la unidad gestora.

- `categoria econômica`: Clasificación económica de los ingresos (por ejemplo, "Receitas Correntes").

- `origem receita`: Fuente específica del ingreso (por ejemplo, "Outras Receitas Correntes").

- `espécie receita`: Tipo de ingreso dentro de la fuente (por ejemplo, "Demais receitas correntes").

- `detalhamento`: Detalle adicional del tipo de ingreso (por ejemplo, "Receita de honorários de advogados").

- `valor previsto atualizado`: Monto actualizado del ingreso previsto (formato texto).

- `valor lançado`: Monto que fue registrado como lanzado (formato texto).

- `valor realizado`: Monto realmente recaudado (formato texto).

- `percentual realizado`: Porcentaje de ejecución respecto al valor previsto.

- `data lançamento`: Fecha en la que se registró la ejecución del ingreso.

- `ano exercício`: Año correspondiente a la ejecución de los ingresos.

In [113]:
import pandas as pd
import numpy as np

In [3]:
from src import soporte_limpieza as sl

In [4]:
pd.set_option("display.max_columns", None)

# 2013

In [5]:
df_2013 = pd.read_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2013.csv",index_col=0, sep=';', encoding='latin-1')

df_2013.head(1)


NOME ÓRGÃO SUPERIOR  CÓDIGO ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                           
63000                  Advocacia-Geral da União         63000   

                                                              NOME ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                  Advocacia-Geral da União - Unidades com víncul...   

                       CÓDIGO UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                           
63000                                  110060   

                                                NOME UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

                      CATEGORIA ECONÔMICA             ORIGEM RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  Receitas Correntes  Outras Receitas Correntes   

                                                     ESPÉCIE RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  Bens, Direitos e Valores Incorporados ao Patr   

                                                        DETALHAMENTO  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  REC.DIVIDA ATIVA NAO TRIBUTARIA DE OUTRAS REC   

                      VALOR PREVISTO ATUALIZADO VALOR LANÇADO VALOR REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                                           
63000                                      0,00          0,00         1297,13   

                      PERCENTUAL REALIZADO DATA LANÇAMENTO  ANO EXERCÍCIO  
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                                 0,00      31/12/2013           2013

In [6]:
df_2013.columns

Index(['NOME ÓRGÃO SUPERIOR', 'CÓDIGO ÓRGÃO', 'NOME ÓRGÃO',
       'CÓDIGO UNIDADE GESTORA', 'NOME UNIDADE GESTORA', 'CATEGORIA ECONÔMICA',
       'ORIGEM RECEITA', 'ESPÉCIE RECEITA', 'DETALHAMENTO',
       'VALOR PREVISTO ATUALIZADO', 'VALOR LANÇADO', 'VALOR REALIZADO',
       'PERCENTUAL REALIZADO', 'DATA LANÇAMENTO', 'ANO EXERCÍCIO'],
      dtype='object')

In [7]:
#Cambiamos el tipo de dato de valor previsto actualizado de objtect a float. Como algún dato está omo 0,00 y las comas las lee como string, cambiamos las , por .
df_2013["VALOR PREVISTO ATUALIZADO"] = df_2013["VALOR PREVISTO ATUALIZADO"].str.replace(',', '.')
df_2013["VALOR PREVISTO ATUALIZADO"] = df_2013["VALOR PREVISTO ATUALIZADO"].astype(float)


In [8]:
df_2013["VALOR LANÇADO"] = df_2013["VALOR LANÇADO"].str.replace(',', '.')
df_2013["VALOR LANÇADO"] = df_2013["VALOR LANÇADO"].astype(float)

In [9]:
df_2013["VALOR REALIZADO"] = df_2013["VALOR REALIZADO"].str.replace(',', '.')
df_2013["VALOR REALIZADO"] = df_2013["VALOR REALIZADO"].astype(float)

In [10]:
df_2013["PERCENTUAL REALIZADO"] = df_2013["PERCENTUAL REALIZADO"].str.replace(',', '.')
df_2013["PERCENTUAL REALIZADO"] = df_2013["PERCENTUAL REALIZADO"].astype(float)

In [11]:
df_2013["DATA LANÇAMENTO"] = pd.to_datetime(df_2013["DATA LANÇAMENTO"], format="%d/%m/%Y")

In [12]:
df_2013.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4498 entries, 63000 to 20000
Data columns (total 15 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   NOME ÓRGÃO SUPERIOR        4498 non-null   object        
 1   CÓDIGO ÓRGÃO               4498 non-null   int64         
 2   NOME ÓRGÃO                 4498 non-null   object        
 3   CÓDIGO UNIDADE GESTORA     4498 non-null   int64         
 4   NOME UNIDADE GESTORA       4498 non-null   object        
 5   CATEGORIA ECONÔMICA        4498 non-null   object        
 6   ORIGEM RECEITA             4498 non-null   object        
 7   ESPÉCIE RECEITA            4498 non-null   object        
 8   DETALHAMENTO               4498 non-null   object        
 9   VALOR PREVISTO ATUALIZADO  4498 non-null   float64       
 10  VALOR LANÇADO              4498 non-null   float64       
 11  VALOR REALIZADO            4498 non-null   float64       
 12  PERCEN

In [13]:
df_2013.sample(2)

NOME ÓRGÃO SUPERIOR  CÓDIGO ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                         
26000                  Ministério da Educação         26234   
26000                  Ministério da Educação         26291   

                                                              NOME ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
26000                             Universidade Federal do Espírito Santo   
26000                  Fundação Coordenação de Aperfeiçoamento de Pes...   

                       CÓDIGO UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                           
26000                                  153046   
26000                                  154003   

                                                NOME UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
26000                         UNIVERSIDADE FEDERAL DO ESPIRITO SANTO   
26000                  FUND.COORD.DE APERF.DE PESSOAL NIVEL SUPERIOR   

                       CATEGORIA ECONÔMICA             ORIGEM RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                   
26000                  Receitas de Capital          Alienação de Bens   
26000                   Receitas Correntes  Outras Receitas Correntes   

                                                   ESPÉCIE RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                
26000                                     Alienação de bens móveis   
26000                  Indenizações, restituições e ressarcimentos   

                                                       DETALHAMENTO  \
CÓDIGO ÓRGÃO SUPERIOR                                                 
26000                               ALIENACAO DE OUTROS BENS MOVEIS   
26000                  RESTITUICOES DE BENEFICIOS NAO DESEMBOLSADOS   

                       VALOR PREVISTO ATUALIZADO  VALOR LANÇADO  \
CÓDIGO ÓRGÃO SUPERIOR                                             
26000                                        0.0            0.0   
26000                                  4678703.0            0.0   

                       VALOR REALIZADO  PERCENTUAL REALIZADO DATA LANÇAMENTO  \
CÓDIGO ÓRGÃO SUPERIOR                                                          
26000                         31330.00                   0.0      2013-12-31   
26000                       5509272.25                 118.0      2013-12-31   

                       ANO EXERCÍCIO  
CÓDIGO ÓRGÃO SUPERIOR                 
26000                           2013  
26000                           2013

In [14]:
traducciones = {
    'NOME ÓRGÃO SUPERIOR': 'NOMBRE ÓRGANO SUPERIOR',
    'CÓDIGO ÓRGÃO': 'CÓDIGO ÓRGANO',
    'NOME ÓRGÃO': 'NOMBRE ÓRGANO',
    'CÓDIGO UNIDADE GESTORA': 'CÓDIGO UNIDAD GESTORA',
    'NOME UNIDADE GESTORA': 'NOMBRE UNIDAD GESTORA',
    'CATEGORIA ECONÔMICA': 'CATEGORÍA ECONÓMICA',
    'ORIGEM RECEITA': 'ORIGEN INGRESO',
    'ESPÉCIE RECEITA': 'TIPO INGRESO',
    'DETALHAMENTO': 'DETALLE',
    'VALOR PREVISTO ATUALIZADO': 'VALOR PREVISTO ACTUALIZADO',
    'VALOR LANÇADO': 'VALOR LANZADO',
    'VALOR REALIZADO': 'VALOR REALIZADO',
    'PERCENTUAL REALIZADO': 'PORCENTAJE REALIZADO',
    'DATA LANÇAMENTO': 'FECHA LANZAMIENTO',
    'ANO EXERCÍCIO': 'AÑO EJERCICIO'
}

df_2013 = df_2013.rename(columns=traducciones)

In [15]:
df_2013.head(1)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                            
63000                  Advocacia-Geral da União          63000   

                                                           NOMBRE ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                  Advocacia-Geral da União - Unidades com víncul...   

                       CÓDIGO UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                          
63000                                 110060   

                                               NOMBRE UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

                      CATEGORÍA ECONÓMICA             ORIGEN INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  Receitas Correntes  Outras Receitas Correntes   

                                                        TIPO INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  Bens, Direitos e Valores Incorporados ao Patr   

                                                             DETALLE  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  REC.DIVIDA ATIVA NAO TRIBUTARIA DE OUTRAS REC   

                       VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                              
63000                                         0.0            0.0   

                       VALOR REALIZADO  PORCENTAJE REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
63000                          1297.13                   0.0   

                      FECHA LANZAMIENTO  AÑO EJERCICIO  
CÓDIGO ÓRGÃO SUPERIOR                                   
63000                        2013-12-31           2013

### NULOS

In [16]:
sl.reporte(df_2013)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0               0.0          object
CÓDIGO ÓRGANO                          0               0.0           int64
NOMBRE ÓRGANO                          0               0.0          object
CÓDIGO UNIDAD GESTORA                  0               0.0           int64
NOMBRE UNIDAD GESTORA                  0               0.0          object
CATEGORÍA ECONÓMICA                    0               0.0          object
ORIGEN INGRESO                         0               0.0          object
TIPO INGRESO                           0               0.0          object
DETALLE                                0               0.0          object
VALOR PREVISTO ACTUALIZADO             0               0.0         float64
VALOR LANZADO                          0               0.0         float64
VALOR REALIZADO                        0               0.0         float64
PORCENTAJE REALIZADO                   0               0.0         float64
FECHA LANZAMIENTO                      0               0.0  datetime64[ns]
AÑO EJERCICIO                          0               0.0           int64

**Manejo de nulos**

En el caso de la base de datos de 2013 no hay nulos que tratar.

In [17]:
df_2013.to_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2013_limpio.csv", index=False, encoding='latin-1')

In [18]:
df_2013.sample(1)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                         
52000                   Ministério da Defesa          52132   

                                          NOMBRE ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                     
52000                  Comando da Marinha - Fundo Naval   

                       CÓDIGO UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                          
52000                                 673001   

                                              NOMBRE UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                 
52000                  DIRETORIA DE FINANCAS-SISTEMA PARA O PAIS/MM   

                      CATEGORÍA ECONÓMICA             ORIGEN INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
52000                  Receitas Correntes  Outras Receitas Correntes   

                                    TIPO INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                              
52000                  Demais receitas correntes   

                                                            DETALLE  \
CÓDIGO ÓRGÃO SUPERIOR                                                 
52000                  COTA-PARTE DO SERV.NO FUNDO DE SAUDE MILITAR   

                       VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                              
52000                                 153780000.0            0.0   

                       VALOR REALIZADO  PORCENTAJE REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
52000                     1.707103e+08                 111.0   

                      FECHA LANZAMIENTO  AÑO EJERCICIO  
CÓDIGO ÓRGÃO SUPERIOR                                   
52000                        2013-12-31           2013

# 2014

In [19]:
df_2014 = pd.read_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2014.csv",index_col=0, sep=';', encoding='latin-1')

df_2014.head(1)

NOME ÓRGÃO SUPERIOR  CÓDIGO ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                           
63000                  Advocacia-Geral da União         63000   

                                                              NOME ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                  Advocacia-Geral da União - Unidades com víncul...   

                       CÓDIGO UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                           
63000                                  110060   

                                                NOME UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

                      CATEGORIA ECONÔMICA             ORIGEM RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  Receitas Correntes  Outras Receitas Correntes   

                                 ESPÉCIE RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                              
63000                  Demais receitas correntes   

                                             DETALHAMENTO  \
CÓDIGO ÓRGÃO SUPERIOR                                       
63000                  RECEITA DE HONORÁRIOS DE ADVOGADOS   

                      VALOR PREVISTO ATUALIZADO VALOR LANÇADO VALOR REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                                           
63000                                      0,00          0,00     46699948,08   

                      PERCENTUAL REALIZADO DATA LANÇAMENTO  ANO EXERCÍCIO  
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                                 0,00      31/12/2014           2014

In [20]:
df_2014["VALOR PREVISTO ATUALIZADO"] = df_2014["VALOR PREVISTO ATUALIZADO"].str.replace(',', '.')
df_2014["VALOR PREVISTO ATUALIZADO"] = df_2014["VALOR PREVISTO ATUALIZADO"].astype(float)

In [21]:
df_2014["VALOR LANÇADO"] = df_2014["VALOR LANÇADO"].str.replace(',', '.')
df_2014["VALOR LANÇADO"] = df_2014["VALOR LANÇADO"].astype(float)

In [22]:
df_2014["VALOR REALIZADO"] = df_2014["VALOR REALIZADO"].str.replace(',', '.')
df_2014["VALOR REALIZADO"] = df_2014["VALOR REALIZADO"].astype(float)

In [23]:
df_2014["PERCENTUAL REALIZADO"] = df_2014["PERCENTUAL REALIZADO"].str.replace(',', '.')
df_2014["PERCENTUAL REALIZADO"] = df_2014["PERCENTUAL REALIZADO"].astype(float)

In [24]:
df_2014["DATA LANÇAMENTO"] = pd.to_datetime(df_2014["DATA LANÇAMENTO"], format="%d/%m/%Y")

In [25]:
df_2014.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4553 entries, 63000 to 20000
Data columns (total 15 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   NOME ÓRGÃO SUPERIOR        4553 non-null   object        
 1   CÓDIGO ÓRGÃO               4553 non-null   int64         
 2   NOME ÓRGÃO                 4553 non-null   object        
 3   CÓDIGO UNIDADE GESTORA     4553 non-null   int64         
 4   NOME UNIDADE GESTORA       4553 non-null   object        
 5   CATEGORIA ECONÔMICA        4553 non-null   object        
 6   ORIGEM RECEITA             4553 non-null   object        
 7   ESPÉCIE RECEITA            4553 non-null   object        
 8   DETALHAMENTO               4553 non-null   object        
 9   VALOR PREVISTO ATUALIZADO  4553 non-null   float64       
 10  VALOR LANÇADO              4553 non-null   float64       
 11  VALOR REALIZADO            4553 non-null   float64       
 12  PERCEN

In [26]:
traducciones = {
    'NOME ÓRGÃO SUPERIOR': 'NOMBRE ÓRGANO SUPERIOR',
    'CÓDIGO ÓRGÃO': 'CÓDIGO ÓRGANO',
    'NOME ÓRGÃO': 'NOMBRE ÓRGANO',
    'CÓDIGO UNIDADE GESTORA': 'CÓDIGO UNIDAD GESTORA',
    'NOME UNIDADE GESTORA': 'NOMBRE UNIDAD GESTORA',
    'CATEGORIA ECONÔMICA': 'CATEGORÍA ECONÓMICA',
    'ORIGEM RECEITA': 'ORIGEN INGRESO',
    'ESPÉCIE RECEITA': 'TIPO INGRESO',
    'DETALHAMENTO': 'DETALLE',
    'VALOR PREVISTO ATUALIZADO': 'VALOR PREVISTO ACTUALIZADO',
    'VALOR LANÇADO': 'VALOR LANZADO',
    'VALOR REALIZADO': 'VALOR REALIZADO',
    'PERCENTUAL REALIZADO': 'PORCENTAJE REALIZADO',
    'DATA LANÇAMENTO': 'FECHA LANZAMIENTO',
    'ANO EXERCÍCIO': 'AÑO EJERCICIO'
}

df_2014 = df_2014.rename(columns=traducciones)

In [27]:
df_2014.head(1)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                            
63000                  Advocacia-Geral da União          63000   

                                                           NOMBRE ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                  Advocacia-Geral da União - Unidades com víncul...   

                       CÓDIGO UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                          
63000                                 110060   

                                               NOMBRE UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

                      CATEGORÍA ECONÓMICA             ORIGEN INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  Receitas Correntes  Outras Receitas Correntes   

                                    TIPO INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                              
63000                  Demais receitas correntes   

                                                  DETALLE  \
CÓDIGO ÓRGÃO SUPERIOR                                       
63000                  RECEITA DE HONORÁRIOS DE ADVOGADOS   

                       VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                              
63000                                         0.0            0.0   

                       VALOR REALIZADO  PORCENTAJE REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
63000                      46699948.08                   0.0   

                      FECHA LANZAMIENTO  AÑO EJERCICIO  
CÓDIGO ÓRGÃO SUPERIOR                                   
63000                        2014-12-31           2014

### Nulos

In [28]:
sl.reporte(df_2014)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0               0.0          object
CÓDIGO ÓRGANO                          0               0.0           int64
NOMBRE ÓRGANO                          0               0.0          object
CÓDIGO UNIDAD GESTORA                  0               0.0           int64
NOMBRE UNIDAD GESTORA                  0               0.0          object
CATEGORÍA ECONÓMICA                    0               0.0          object
ORIGEN INGRESO                         0               0.0          object
TIPO INGRESO                           0               0.0          object
DETALLE                                0               0.0          object
VALOR PREVISTO ACTUALIZADO             0               0.0         float64
VALOR LANZADO                          0               0.0         float64
VALOR REALIZADO                        0               0.0         float64
PORCENTAJE REALIZADO                   0               0.0         float64
FECHA LANZAMIENTO                      0               0.0  datetime64[ns]
AÑO EJERCICIO                          0               0.0           int64

**Manejo de nulos**

En el caso de la base de datos de 2014 no hay nulos que tratar.

In [29]:
df_2014.to_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2014_limpio.csv", index=False, encoding='latin-1')

In [30]:
df_2014.sample(1)

NOMBRE ÓRGANO SUPERIOR  \
CÓDIGO ÓRGÃO SUPERIOR                                              
30000                  Ministério da Justiça e Segurança Pública   

                       CÓDIGO ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                  
30000                          30802   

                                                    NOMBRE ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                               
30000                  Departamento de Polícia Rodoviária Federal   

                       CÓDIGO UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                          
30000                                 200109   

                                     NOMBRE UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                        
30000                  DEPTO.DE POLICIA RODOVIARIA FEDERAL   

                      CATEGORÍA ECONÓMICA       ORIGEN INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                            
30000                  Receitas Correntes  Receita de Serviços   

                              TIPO INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                        
30000                  Receita de Serviços   

                                                         DETALLE  \
CÓDIGO ÓRGÃO SUPERIOR                                              
30000                  TAR.INSCR.CONCURSOS E PROCESSOS SELETIVOS   

                       VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                              
30000                                         0.0            0.0   

                       VALOR REALIZADO  PORCENTAJE REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
30000                      11720125.41                   0.0   

                      FECHA LANZAMIENTO  AÑO EJERCICIO  
CÓDIGO ÓRGÃO SUPERIOR                                   
30000                        2014-12-31           2014

# 2015

In [31]:
df_2015 = pd.read_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2015.csv",index_col=0, sep=';', encoding='latin-1')

df_2015.sample(1)

NOME ÓRGÃO SUPERIOR  CÓDIGO ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                                         
53000                  Ministério do Desenvolvimento Regional         53901   

                                                           NOME ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                                   
53000                  Fundo Constitucional de Financiamento do Norte   

                       CÓDIGO UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                           
53000                                  537002   

                                           NOME UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                             
53000                  FUNDO CONSTITUCIONAL FINANCIAMENTO NORTE   

                      CATEGORIA ECONÔMICA             ORIGEM RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
53000                  Receitas Correntes  Outras Receitas Correntes   

                                                     ESPÉCIE RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
53000                  Multas administrativas, contratuais e judicia   

                                                      DETALHAMENTO  \
CÓDIGO ÓRGÃO SUPERIOR                                                
53000                  REC DE PARC - MULTA E JUROS DE MORA DO IRPF   

                      VALOR PREVISTO ATUALIZADO VALOR LANÇADO VALOR REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                                           
53000                                 688577,00          0,00            0,00   

                      PERCENTUAL REALIZADO DATA LANÇAMENTO  ANO EXERCÍCIO  
CÓDIGO ÓRGÃO SUPERIOR                                                      
53000                                 0,00      31/12/2015           2015

In [32]:
df_2015["VALOR PREVISTO ATUALIZADO"] = df_2015["VALOR PREVISTO ATUALIZADO"].str.replace(',', '.')
df_2015["VALOR PREVISTO ATUALIZADO"] = df_2015["VALOR PREVISTO ATUALIZADO"].astype(float)

In [33]:
df_2015["VALOR LANÇADO"] = df_2015["VALOR LANÇADO"].str.replace(',', '.')
df_2015["VALOR LANÇADO"] = df_2015["VALOR LANÇADO"].astype(float)

In [34]:
df_2015["VALOR REALIZADO"] = df_2015["VALOR REALIZADO"].str.replace(',', '.')
df_2015["VALOR REALIZADO"] = df_2015["VALOR REALIZADO"].astype(float)

In [35]:
df_2015["PERCENTUAL REALIZADO"] = df_2015["PERCENTUAL REALIZADO"].str.replace(',', '.')
df_2015["PERCENTUAL REALIZADO"] = df_2015["PERCENTUAL REALIZADO"].astype(float)

In [36]:
df_2015["DATA LANÇAMENTO"] = pd.to_datetime(df_2015["DATA LANÇAMENTO"], format="%d/%m/%Y")

In [37]:
df_2015.info()

<class 'pandas.core.frame.DataFrame'>
Index: 4523 entries, 63000 to 20000
Data columns (total 15 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   NOME ÓRGÃO SUPERIOR        4523 non-null   object        
 1   CÓDIGO ÓRGÃO               4523 non-null   int64         
 2   NOME ÓRGÃO                 4523 non-null   object        
 3   CÓDIGO UNIDADE GESTORA     4523 non-null   int64         
 4   NOME UNIDADE GESTORA       4523 non-null   object        
 5   CATEGORIA ECONÔMICA        4523 non-null   object        
 6   ORIGEM RECEITA             4523 non-null   object        
 7   ESPÉCIE RECEITA            4523 non-null   object        
 8   DETALHAMENTO               4523 non-null   object        
 9   VALOR PREVISTO ATUALIZADO  4523 non-null   float64       
 10  VALOR LANÇADO              4523 non-null   float64       
 11  VALOR REALIZADO            4523 non-null   float64       
 12  PERCEN

In [38]:
traducciones = {
    'NOME ÓRGÃO SUPERIOR': 'NOMBRE ÓRGANO SUPERIOR',
    'CÓDIGO ÓRGÃO': 'CÓDIGO ÓRGANO',
    'NOME ÓRGÃO': 'NOMBRE ÓRGANO',
    'CÓDIGO UNIDADE GESTORA': 'CÓDIGO UNIDAD GESTORA',
    'NOME UNIDADE GESTORA': 'NOMBRE UNIDAD GESTORA',
    'CATEGORIA ECONÔMICA': 'CATEGORÍA ECONÓMICA',
    'ORIGEM RECEITA': 'ORIGEN INGRESO',
    'ESPÉCIE RECEITA': 'TIPO INGRESO',
    'DETALHAMENTO': 'DETALLE',
    'VALOR PREVISTO ATUALIZADO': 'VALOR PREVISTO ACTUALIZADO',
    'VALOR LANÇADO': 'VALOR LANZADO',
    'VALOR REALIZADO': 'VALOR REALIZADO',
    'PERCENTUAL REALIZADO': 'PORCENTAJE REALIZADO',
    'DATA LANÇAMENTO': 'FECHA LANZAMIENTO',
    'ANO EXERCÍCIO': 'AÑO EJERCICIO'
}

df_2015 = df_2015.rename(columns=traducciones)

In [39]:
sl.reporte(df_2015)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0               0.0          object
CÓDIGO ÓRGANO                          0               0.0           int64
NOMBRE ÓRGANO                          0               0.0          object
CÓDIGO UNIDAD GESTORA                  0               0.0           int64
NOMBRE UNIDAD GESTORA                  0               0.0          object
CATEGORÍA ECONÓMICA                    0               0.0          object
ORIGEN INGRESO                         0               0.0          object
TIPO INGRESO                           0               0.0          object
DETALLE                                0               0.0          object
VALOR PREVISTO ACTUALIZADO             0               0.0         float64
VALOR LANZADO                          0               0.0         float64
VALOR REALIZADO                        0               0.0         float64
PORCENTAJE REALIZADO                   0               0.0         float64
FECHA LANZAMIENTO                      0               0.0  datetime64[ns]
AÑO EJERCICIO                          0               0.0           int64

**Manejo de nulos**

En el caso de la base de datos de 2015 no hay nulos que tratar.

In [40]:
df_2015.to_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2015_limpio.csv", index=False, encoding='latin-1')

In [41]:
df_2015.sample(1)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                               
44000                  Ministério do Meio Ambiente          44000   

                                                           NOMBRE ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
44000                  Ministério do Meio Ambiente - Unidades com vín...   

                       CÓDIGO UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                          
44000                                 440002   

                                               NOMBRE UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
44000                  SUBSECRET.PLANEJAM.ORCAM.E ADMINIST±SPOA(MMA)   

                      CATEGORÍA ECONÓMICA       ORIGEN INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                            
44000                  Receitas Correntes  Receita Patrimonial   

                                          TIPO INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                    
44000                  Exploração de recursos naturais   

                                                       DETALLE  \
CÓDIGO ÓRGÃO SUPERIOR                                            
44000                  UTIL.RECURSO HIDRICOS - DEMAIS EMPRESAS   

                       VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                              
44000                                  64927112.0            0.0   

                       VALOR REALIZADO  PORCENTAJE REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
44000                      44629706.33                  69.0   

                      FECHA LANZAMIENTO  AÑO EJERCICIO  
CÓDIGO ÓRGÃO SUPERIOR                                   
44000                        2015-12-31           2015

# 2016

In [42]:
df_2016 = pd.read_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2016.csv",index_col=0, sep=';', encoding='latin-1')

df_2016.head(1)

NOME ÓRGÃO SUPERIOR  CÓDIGO ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                           
63000                  Advocacia-Geral da União         63000   

                                                              NOME ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                  Advocacia-Geral da União - Unidades com víncul...   

                       CÓDIGO UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                           
63000                                  110060   

                                                NOME UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

                      CATEGORIA ECONÔMICA             ORIGEM RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  Receitas Correntes  Outras Receitas Correntes   

                                 ESPÉCIE RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                              
63000                  Demais receitas correntes   

                                                        DETALHAMENTO  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  ENCARGOS LEGAIS PELA INSCR.EM DIV.ATIVA-PRINC   

                      VALOR PREVISTO ATUALIZADO VALOR LANÇADO VALOR REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                                           
63000                                      0,00          0,00       154948,98   

                      PERCENTUAL REALIZADO DATA LANÇAMENTO  ANO EXERCÍCIO  
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                                 0,00      04/04/2016           2016

In [43]:
df_2016["VALOR PREVISTO ATUALIZADO"] = df_2016["VALOR PREVISTO ATUALIZADO"].str.replace(',', '.')
df_2016["VALOR PREVISTO ATUALIZADO"] = df_2016["VALOR PREVISTO ATUALIZADO"].astype(float)

In [44]:
df_2016["VALOR LANÇADO"] = df_2016["VALOR LANÇADO"].str.replace(',', '.')
df_2016["VALOR LANÇADO"] = df_2016["VALOR LANÇADO"].astype(float)

In [45]:
df_2016["VALOR REALIZADO"] = df_2016["VALOR REALIZADO"].str.replace(',', '.')
df_2016["VALOR REALIZADO"] = df_2016["VALOR REALIZADO"].astype(float)

In [46]:
df_2016["PERCENTUAL REALIZADO"] = df_2016["PERCENTUAL REALIZADO"].str.replace(',', '.')
df_2016["PERCENTUAL REALIZADO"] = df_2016["PERCENTUAL REALIZADO"].astype(float)

In [47]:
df_2016["DATA LANÇAMENTO"] = pd.to_datetime(df_2016["DATA LANÇAMENTO"], format="%d/%m/%Y")

In [48]:
df_2016.info()

<class 'pandas.core.frame.DataFrame'>
Index: 194533 entries, 63000 to 20000
Data columns (total 15 columns):
 #   Column                     Non-Null Count   Dtype         
---  ------                     --------------   -----         
 0   NOME ÓRGÃO SUPERIOR        194533 non-null  object        
 1   CÓDIGO ÓRGÃO               194533 non-null  int64         
 2   NOME ÓRGÃO                 194533 non-null  object        
 3   CÓDIGO UNIDADE GESTORA     194533 non-null  int64         
 4   NOME UNIDADE GESTORA       194533 non-null  object        
 5   CATEGORIA ECONÔMICA        194533 non-null  object        
 6   ORIGEM RECEITA             194533 non-null  object        
 7   ESPÉCIE RECEITA            194533 non-null  object        
 8   DETALHAMENTO               194533 non-null  object        
 9   VALOR PREVISTO ATUALIZADO  194533 non-null  float64       
 10  VALOR LANÇADO              194533 non-null  float64       
 11  VALOR REALIZADO            194533 non-null  float64   

In [49]:
traducciones = {
    'NOME ÓRGÃO SUPERIOR': 'NOMBRE ÓRGANO SUPERIOR',
    'CÓDIGO ÓRGÃO': 'CÓDIGO ÓRGANO',
    'NOME ÓRGÃO': 'NOMBRE ÓRGANO',
    'CÓDIGO UNIDADE GESTORA': 'CÓDIGO UNIDAD GESTORA',
    'NOME UNIDADE GESTORA': 'NOMBRE UNIDAD GESTORA',
    'CATEGORIA ECONÔMICA': 'CATEGORÍA ECONÓMICA',
    'ORIGEM RECEITA': 'ORIGEN INGRESO',
    'ESPÉCIE RECEITA': 'TIPO INGRESO',
    'DETALHAMENTO': 'DETALLE',
    'VALOR PREVISTO ATUALIZADO': 'VALOR PREVISTO ACTUALIZADO',
    'VALOR LANÇADO': 'VALOR LANZADO',
    'VALOR REALIZADO': 'VALOR REALIZADO',
    'PERCENTUAL REALIZADO': 'PORCENTAJE REALIZADO',
    'DATA LANÇAMENTO': 'FECHA LANZAMIENTO',
    'ANO EXERCÍCIO': 'AÑO EJERCICIO'
}

df_2016 = df_2016.rename(columns=traducciones)

In [50]:
sl.reporte(df_2016)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0              0.00          object
CÓDIGO ÓRGANO                          0              0.00           int64
NOMBRE ÓRGANO                          0              0.00          object
CÓDIGO UNIDAD GESTORA                  0              0.00           int64
NOMBRE UNIDAD GESTORA                  0              0.00          object
CATEGORÍA ECONÓMICA                    0              0.00          object
ORIGEN INGRESO                         0              0.00          object
TIPO INGRESO                           0              0.00          object
DETALLE                                0              0.00          object
VALOR PREVISTO ACTUALIZADO             0              0.00         float64
VALOR LANZADO                          0              0.00         float64
VALOR REALIZADO                        0              0.00         float64
PORCENTAJE REALIZADO                   0              0.00         float64
FECHA LANZAMIENTO                     82              0.04  datetime64[ns]
AÑO EJERCICIO                          0              0.00           int64

In [51]:
df_2016.shape

(194533, 15)

**Manejo de nulos**

En el caso de la base de datos de 2016 vemos que hay un total de 82 nulos en la columna "FECHA DE LANZAMIENTO". Al tener casi 200.000 datos y corresponde a un 0,04% respecto al total de los datos, es factible rellenar los datos nulos con la fecha media para poder cubrir las celdas que no tienen datos, ya que no afectará significativamente al análisis.

In [52]:
pd.DataFrame(df_2016["FECHA LANZAMIENTO"].describe()).T

count                           mean                  min  \
FECHA LANZAMIENTO  194451  2016-06-29 04:03:52.605643520  2016-01-01 00:00:00   

                                   25%                  50%  \
FECHA LANZAMIENTO  2016-03-30 00:00:00  2016-06-29 00:00:00   

                                   75%                  max  
FECHA LANZAMIENTO  2016-09-29 00:00:00  2016-12-31 00:00:00

In [53]:
df_2016["FECHA LANZAMIENTO"] = df_2016["FECHA LANZAMIENTO"].fillna(df_2016["FECHA LANZAMIENTO"].mean())


In [54]:
sl.reporte(df_2016)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0               0.0          object
CÓDIGO ÓRGANO                          0               0.0           int64
NOMBRE ÓRGANO                          0               0.0          object
CÓDIGO UNIDAD GESTORA                  0               0.0           int64
NOMBRE UNIDAD GESTORA                  0               0.0          object
CATEGORÍA ECONÓMICA                    0               0.0          object
ORIGEN INGRESO                         0               0.0          object
TIPO INGRESO                           0               0.0          object
DETALLE                                0               0.0          object
VALOR PREVISTO ACTUALIZADO             0               0.0         float64
VALOR LANZADO                          0               0.0         float64
VALOR REALIZADO                        0               0.0         float64
PORCENTAJE REALIZADO                   0               0.0         float64
FECHA LANZAMIENTO                      0               0.0  datetime64[ns]
AÑO EJERCICIO                          0               0.0           int64

In [55]:
df_2016.to_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2016_limpio.csv", index=False, encoding='latin-1')

In [56]:
df_2016.sample(1)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                                
39000                  Ministério da Infraestrutura          39250   

                                                    NOMBRE ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                               
39000                  Agência Nacional de Transportes Terrestres   

                       CÓDIGO UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                          
39000                                 393001   

                                            NOMBRE UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                               
39000                  AGENCIA NACIONAL DE TRANSPORTES TERRESTRES   

                      CATEGORÍA ECONÓMICA                      ORIGEN INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                                           
39000                  Receitas Correntes  Receitas Correntes - a classificar   

                                             TIPO INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                       
39000                  Receitas Correntes - a classificar   

                                                  DETALLE  \
CÓDIGO ÓRGÃO SUPERIOR                                       
39000                  Receitas Correntes - a classificar   

                       VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                              
39000                                         0.0            0.0   

                       VALOR REALIZADO  PORCENTAJE REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
39000                            48.58                   0.0   

                      FECHA LANZAMIENTO  AÑO EJERCICIO  
CÓDIGO ÓRGÃO SUPERIOR                                   
39000                        2016-03-14           2016

# 2017

In [57]:
df_2017 = pd.read_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2017.csv",index_col=0, sep=';', encoding='latin-1')

df_2017.head(1)

NOME ÓRGÃO SUPERIOR  CÓDIGO ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                           
63000                  Advocacia-Geral da União         63000   

                                                              NOME ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                  Advocacia-Geral da União - Unidades com víncul...   

                       CÓDIGO UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                           
63000                                  110060   

                                                NOME UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

                      CATEGORIA ECONÔMICA             ORIGEM RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  Receitas Correntes  Outras Receitas Correntes   

                                 ESPÉCIE RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                              
63000                  Demais receitas correntes   

                                              DETALHAMENTO  \
CÓDIGO ÓRGÃO SUPERIOR                                        
63000                  OUTRAS RECEITAS-PRIMARIAS-PRINCIPAL   

                      VALOR PREVISTO ATUALIZADO VALOR LANÇADO VALOR REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                                           
63000                                      0,00          0,00          198,00   

                      PERCENTUAL REALIZADO DATA LANÇAMENTO  ANO EXERCÍCIO  
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                                 0,00      29/08/2017           2017

In [58]:
df_2017["VALOR PREVISTO ATUALIZADO"] = df_2017["VALOR PREVISTO ATUALIZADO"].str.replace(',', '.')
df_2017["VALOR PREVISTO ATUALIZADO"] = df_2017["VALOR PREVISTO ATUALIZADO"].astype(float)

In [59]:
df_2017["VALOR LANÇADO"] = df_2017["VALOR LANÇADO"].str.replace(',', '.')
df_2017["VALOR LANÇADO"] = df_2017["VALOR LANÇADO"].astype(float)

In [60]:
df_2017["VALOR REALIZADO"] = df_2017["VALOR REALIZADO"].str.replace(',', '.')
df_2017["VALOR REALIZADO"] = df_2017["VALOR REALIZADO"].astype(float)

In [61]:
df_2017["PERCENTUAL REALIZADO"] = df_2017["PERCENTUAL REALIZADO"].str.replace(',', '.')
df_2017["PERCENTUAL REALIZADO"] = df_2017["PERCENTUAL REALIZADO"].astype(float)

In [62]:
df_2017["DATA LANÇAMENTO"] = pd.to_datetime(df_2017["DATA LANÇAMENTO"], format="%d/%m/%Y")

In [63]:
df_2017.info()

<class 'pandas.core.frame.DataFrame'>
Index: 190479 entries, 63000 to 20000
Data columns (total 15 columns):
 #   Column                     Non-Null Count   Dtype         
---  ------                     --------------   -----         
 0   NOME ÓRGÃO SUPERIOR        190479 non-null  object        
 1   CÓDIGO ÓRGÃO               190479 non-null  int64         
 2   NOME ÓRGÃO                 190479 non-null  object        
 3   CÓDIGO UNIDADE GESTORA     190479 non-null  int64         
 4   NOME UNIDADE GESTORA       190479 non-null  object        
 5   CATEGORIA ECONÔMICA        190479 non-null  object        
 6   ORIGEM RECEITA             190479 non-null  object        
 7   ESPÉCIE RECEITA            190479 non-null  object        
 8   DETALHAMENTO               190479 non-null  object        
 9   VALOR PREVISTO ATUALIZADO  190479 non-null  float64       
 10  VALOR LANÇADO              190479 non-null  float64       
 11  VALOR REALIZADO            190479 non-null  float64   

In [64]:
traducciones = {
    'NOME ÓRGÃO SUPERIOR': 'NOMBRE ÓRGANO SUPERIOR',
    'CÓDIGO ÓRGÃO': 'CÓDIGO ÓRGANO',
    'NOME ÓRGÃO': 'NOMBRE ÓRGANO',
    'CÓDIGO UNIDADE GESTORA': 'CÓDIGO UNIDAD GESTORA',
    'NOME UNIDADE GESTORA': 'NOMBRE UNIDAD GESTORA',
    'CATEGORIA ECONÔMICA': 'CATEGORÍA ECONÓMICA',
    'ORIGEM RECEITA': 'ORIGEN INGRESO',
    'ESPÉCIE RECEITA': 'TIPO INGRESO',
    'DETALHAMENTO': 'DETALLE',
    'VALOR PREVISTO ATUALIZADO': 'VALOR PREVISTO ACTUALIZADO',
    'VALOR LANÇADO': 'VALOR LANZADO',
    'VALOR REALIZADO': 'VALOR REALIZADO',
    'PERCENTUAL REALIZADO': 'PORCENTAJE REALIZADO',
    'DATA LANÇAMENTO': 'FECHA LANZAMIENTO',
    'ANO EXERCÍCIO': 'AÑO EJERCICIO'
}

df_2017 = df_2017.rename(columns=traducciones)

### Nulos

In [65]:
sl.reporte(df_2017)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0              0.00          object
CÓDIGO ÓRGANO                          0              0.00           int64
NOMBRE ÓRGANO                          0              0.00          object
CÓDIGO UNIDAD GESTORA                  0              0.00           int64
NOMBRE UNIDAD GESTORA                  0              0.00          object
CATEGORÍA ECONÓMICA                    0              0.00          object
ORIGEN INGRESO                         0              0.00          object
TIPO INGRESO                           0              0.00          object
DETALLE                                0              0.00          object
VALOR PREVISTO ACTUALIZADO             0              0.00         float64
VALOR LANZADO                          0              0.00         float64
VALOR REALIZADO                        0              0.00         float64
PORCENTAJE REALIZADO                   0              0.00         float64
FECHA LANZAMIENTO                     86              0.05  datetime64[ns]
AÑO EJERCICIO                          0              0.00           int64

In [66]:
df_2017.shape

(190479, 15)

**Manejo de nulos**

En el caso de la base de datos de 2017 vemos que hay un total de 82 nulos en la columna "FECHA DE LANZAMIENTO". Al tener casi 200.000 datos y corresponder a un 0,05% respecto al total de los datos, es factible rellenar los datos nulos con la fecha media para poder cubrir las celdas que no tienen datos, ya que no afectará significativamente al análisis.

In [67]:
df_2017["FECHA LANZAMIENTO"] = df_2017["FECHA LANZAMIENTO"].fillna(df_2017["FECHA LANZAMIENTO"].mean())

In [68]:
sl.reporte(df_2017)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0               0.0          object
CÓDIGO ÓRGANO                          0               0.0           int64
NOMBRE ÓRGANO                          0               0.0          object
CÓDIGO UNIDAD GESTORA                  0               0.0           int64
NOMBRE UNIDAD GESTORA                  0               0.0          object
CATEGORÍA ECONÓMICA                    0               0.0          object
ORIGEN INGRESO                         0               0.0          object
TIPO INGRESO                           0               0.0          object
DETALLE                                0               0.0          object
VALOR PREVISTO ACTUALIZADO             0               0.0         float64
VALOR LANZADO                          0               0.0         float64
VALOR REALIZADO                        0               0.0         float64
PORCENTAJE REALIZADO                   0               0.0         float64
FECHA LANZAMIENTO                      0               0.0  datetime64[ns]
AÑO EJERCICIO                          0               0.0           int64

In [69]:
df_2017.to_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2017_limpio.csv", index=False, encoding='latin-1')

In [70]:
df_2017.sample(2)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
26000                  Ministério da Educação          26251   
26000                  Ministério da Educação          26243   

                                                     NOMBRE ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                                
26000                   Fundação Universidade Federal do Tocantins   
26000                  Universidade Federal do Rio Grande do Norte   

                       CÓDIGO UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                          
26000                                 154419   
26000                                 153103   

                                             NOMBRE UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                
26000                   FUNDACAO UNIVERSIDADE FEDERAL DO TOCANTINS   
26000                  UNIVERSIDADE FEDERAL DO RIO GRANDE DO NORTE   

                      CATEGORÍA ECONÓMICA       ORIGEN INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                            
26000                  Receitas Correntes  Receita de Serviços   
26000                  Receitas Correntes  Receita Patrimonial   

                                                       TIPO INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                                 
26000                  Serviços Administrativos e Comerciais Gerais   
26000                                           Valores Mobiliários   

                                                            DETALLE  \
CÓDIGO ÓRGÃO SUPERIOR                                                 
26000                   SERV.ADMINISTRAT.E COMERCIAIS GERAIS-PRINC.   
26000                  REMUNERACAO DE DEPOSITOS BANCARIOS-PRINCIPAL   

                       VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                              
26000                                         0.0            0.0   
26000                                         0.0            0.0   

                       VALOR REALIZADO  PORCENTAJE REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
26000                         11222.50                   0.0   
26000                        157125.26                   0.0   

                      FECHA LANZAMIENTO  AÑO EJERCICIO  
CÓDIGO ÓRGÃO SUPERIOR                                   
26000                        2017-06-29           2017  
26000                        2017-07-07           2017

# 2018

In [71]:
df_2018 = pd.read_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2018.csv",index_col=0, sep=';', encoding='latin-1')

df_2018.head(1)

NOME ÓRGÃO SUPERIOR  CÓDIGO ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                           
63000                  Advocacia-Geral da União         63000   

                                                              NOME ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                  Advocacia-Geral da União - Unidades com víncul...   

                       CÓDIGO UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                           
63000                                  110060   

                                                NOME UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

                      CATEGORIA ECONÔMICA                      ORIGEM RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                           
63000                  Receitas Correntes  Receitas Correntes - a classificar   

                                          ESPÉCIE RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                       
63000                  Receitas Correntes - a classificar   

                                             DETALHAMENTO  \
CÓDIGO ÓRGÃO SUPERIOR                                       
63000                  Receitas Correntes - a classificar   

                      VALOR PREVISTO ATUALIZADO VALOR LANÇADO VALOR REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                                           
63000                                      0,00          0,00        -1693,72   

                      PERCENTUAL REALIZADO DATA LANÇAMENTO  ANO EXERCÍCIO  
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                                 0,00      25/07/2018           2018

In [72]:
df_2018["VALOR PREVISTO ATUALIZADO"] = df_2018["VALOR PREVISTO ATUALIZADO"].str.replace(',', '.')
df_2018["VALOR PREVISTO ATUALIZADO"] = df_2018["VALOR PREVISTO ATUALIZADO"].astype(float)

In [73]:
df_2018["VALOR LANÇADO"] = df_2018["VALOR LANÇADO"].str.replace(',', '.')
df_2018["VALOR LANÇADO"] = df_2018["VALOR LANÇADO"].astype(float)

In [74]:
df_2018["VALOR REALIZADO"] = df_2018["VALOR REALIZADO"].str.replace(',', '.')
df_2018["VALOR REALIZADO"] = df_2018["VALOR REALIZADO"].astype(float)

In [75]:
df_2018["PERCENTUAL REALIZADO"] = df_2018["PERCENTUAL REALIZADO"].str.replace(',', '.')
df_2018["PERCENTUAL REALIZADO"] = df_2018["PERCENTUAL REALIZADO"].astype(float)

In [76]:
df_2018["DATA LANÇAMENTO"] = pd.to_datetime(df_2018["DATA LANÇAMENTO"], format="%d/%m/%Y")

In [77]:
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
Index: 173944 entries, 63000 to 20000
Data columns (total 15 columns):
 #   Column                     Non-Null Count   Dtype         
---  ------                     --------------   -----         
 0   NOME ÓRGÃO SUPERIOR        173944 non-null  object        
 1   CÓDIGO ÓRGÃO               173944 non-null  int64         
 2   NOME ÓRGÃO                 173944 non-null  object        
 3   CÓDIGO UNIDADE GESTORA     173944 non-null  int64         
 4   NOME UNIDADE GESTORA       173944 non-null  object        
 5   CATEGORIA ECONÔMICA        173944 non-null  object        
 6   ORIGEM RECEITA             173944 non-null  object        
 7   ESPÉCIE RECEITA            173944 non-null  object        
 8   DETALHAMENTO               173944 non-null  object        
 9   VALOR PREVISTO ATUALIZADO  173944 non-null  float64       
 10  VALOR LANÇADO              173944 non-null  float64       
 11  VALOR REALIZADO            173944 non-null  float64   

In [78]:
traducciones = {
    'NOME ÓRGÃO SUPERIOR': 'NOMBRE ÓRGANO SUPERIOR',
    'CÓDIGO ÓRGÃO': 'CÓDIGO ÓRGANO',
    'NOME ÓRGÃO': 'NOMBRE ÓRGANO',
    'CÓDIGO UNIDADE GESTORA': 'CÓDIGO UNIDAD GESTORA',
    'NOME UNIDADE GESTORA': 'NOMBRE UNIDAD GESTORA',
    'CATEGORIA ECONÔMICA': 'CATEGORÍA ECONÓMICA',
    'ORIGEM RECEITA': 'ORIGEN INGRESO',
    'ESPÉCIE RECEITA': 'TIPO INGRESO',
    'DETALHAMENTO': 'DETALLE',
    'VALOR PREVISTO ATUALIZADO': 'VALOR PREVISTO ACTUALIZADO',
    'VALOR LANÇADO': 'VALOR LANZADO',
    'VALOR REALIZADO': 'VALOR REALIZADO',
    'PERCENTUAL REALIZADO': 'PORCENTAJE REALIZADO',
    'DATA LANÇAMENTO': 'FECHA LANZAMIENTO',
    'ANO EXERCÍCIO': 'AÑO EJERCICIO'
}

df_2018 = df_2018.rename(columns=traducciones)

### Nulos

In [79]:
sl.reporte(df_2018)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0              0.00          object
CÓDIGO ÓRGANO                          0              0.00           int64
NOMBRE ÓRGANO                          0              0.00          object
CÓDIGO UNIDAD GESTORA                  0              0.00           int64
NOMBRE UNIDAD GESTORA                  0              0.00          object
CATEGORÍA ECONÓMICA                    0              0.00          object
ORIGEN INGRESO                         0              0.00          object
TIPO INGRESO                           0              0.00          object
DETALLE                                0              0.00          object
VALOR PREVISTO ACTUALIZADO             0              0.00         float64
VALOR LANZADO                          0              0.00         float64
VALOR REALIZADO                        0              0.00         float64
PORCENTAJE REALIZADO                   0              0.00         float64
FECHA LANZAMIENTO                    115              0.07  datetime64[ns]
AÑO EJERCICIO                          0              0.00           int64

In [80]:
df_2018.shape

(173944, 15)

**Manejo de nulos**

En el caso de la base de datos de 2018 vemos que hay un total de 115 nulos en la columna "FECHA DE LANZAMIENTO". Al tener casi 175.000 datos y corresponder a un 0,07% respecto al total de los datos, es factible rellenar los datos nulos con la fecha media para poder cubrir las celdas que no tienen datos, ya que no afectará significativamente al análisis.

In [81]:
df_2018["FECHA LANZAMIENTO"] = df_2018["FECHA LANZAMIENTO"].fillna(df_2018["FECHA LANZAMIENTO"].mean())

In [82]:
sl.reporte(df_2018)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0               0.0          object
CÓDIGO ÓRGANO                          0               0.0           int64
NOMBRE ÓRGANO                          0               0.0          object
CÓDIGO UNIDAD GESTORA                  0               0.0           int64
NOMBRE UNIDAD GESTORA                  0               0.0          object
CATEGORÍA ECONÓMICA                    0               0.0          object
ORIGEN INGRESO                         0               0.0          object
TIPO INGRESO                           0               0.0          object
DETALLE                                0               0.0          object
VALOR PREVISTO ACTUALIZADO             0               0.0         float64
VALOR LANZADO                          0               0.0         float64
VALOR REALIZADO                        0               0.0         float64
PORCENTAJE REALIZADO                   0               0.0         float64
FECHA LANZAMIENTO                      0               0.0  datetime64[ns]
AÑO EJERCICIO                          0               0.0           int64

In [83]:
df_2018.to_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2018_limpio.csv", index=False, encoding='latin-1')

In [84]:
df_2018.sample(2)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                               
26000                       Ministério da Educação          26249   
41000                  Ministério das Comunicações          41231   

                                                      NOMBRE ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                                 
26000                  Universidade Federal Rural do Rio de Janeiro   
41000                         Agêncial Nacional de Telecomunicações   

                       CÓDIGO UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                          
26000                                 153166   
41000                                 413001   

                                              NOMBRE UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                 
26000                  UNIVERSIDADE FEDERAL RURAL DO RIO DE JANEIRO   
41000                     AGENCIA NACIONAL DE TELECOMUNICACOES-SEDE   

                      CATEGORÍA ECONÓMICA       ORIGEN INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                            
26000                  Receitas Correntes  Receita Patrimonial   
41000                  Receitas Correntes  Receita de Serviços   

                                                        TIPO INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
26000                  Exploração do patrimônio imobiliário do Estad   
41000                   Serviços Administrativos e Comerciais Gerais   

                                                           DETALLE  \
CÓDIGO ÓRGÃO SUPERIOR                                                
26000                           ALUGUEIS E ARRENDAMENTOS-PRINCIPAL   
41000                  SERV.ADMINISTRAT.E COMERCIAIS GERAIS-PRINC.   

                       VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                              
26000                                         0.0            0.0   
41000                                         0.0            0.0   

                       VALOR REALIZADO  PORCENTAJE REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
26000                           442.13                   0.0   
41000                           185.87                   0.0   

                      FECHA LANZAMIENTO  AÑO EJERCICIO  
CÓDIGO ÓRGÃO SUPERIOR                                   
26000                        2018-02-09           2018  
41000                        2018-02-01           2018

# 2019

In [85]:
df_2019 = pd.read_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2019.csv",index_col=0, sep=';', encoding='latin-1')

df_2019.head(1)

NOME ÓRGÃO SUPERIOR  CÓDIGO ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                           
63000                  Advocacia-Geral da União         63000   

                                                              NOME ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                  Advocacia-Geral da União - Unidades com víncul...   

                       CÓDIGO UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                           
63000                                  110060   

                                                NOME UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

                      CATEGORIA ECONÔMICA       ORIGEM RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                            
63000                  Receitas Correntes  Receita de Serviços   

                                                    ESPÉCIE RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                 
63000                  Serviços Administrativos e Comerciais Gerais   

                                                        DETALHAMENTO  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  INSCR.EM CONCURSOS E PROC.SELETIVOS-PRINCIPAL   

                      VALOR PREVISTO ATUALIZADO VALOR LANÇADO VALOR REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                                           
63000                                      0,00          0,00          -95,00   

                      PERCENTUAL REALIZADO DATA LANÇAMENTO  ANO EXERCÍCIO  
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                                 0,00      12/06/2019           2019

In [86]:
df_2019["VALOR PREVISTO ATUALIZADO"] = df_2019["VALOR PREVISTO ATUALIZADO"].str.replace(',', '.')
df_2019["VALOR PREVISTO ATUALIZADO"] = df_2019["VALOR PREVISTO ATUALIZADO"].astype(float)

In [87]:
df_2019["VALOR LANÇADO"] = df_2019["VALOR LANÇADO"].str.replace(',', '.')
df_2019["VALOR LANÇADO"] = df_2019["VALOR LANÇADO"].astype(float)

In [88]:
df_2019["VALOR REALIZADO"] = df_2019["VALOR REALIZADO"].str.replace(',', '.')
df_2019["VALOR REALIZADO"] = df_2019["VALOR REALIZADO"].astype(float)

In [89]:
df_2019["PERCENTUAL REALIZADO"] = df_2019["PERCENTUAL REALIZADO"].str.replace(',', '.')
df_2019["PERCENTUAL REALIZADO"] = df_2019["PERCENTUAL REALIZADO"].astype(float)

In [90]:
df_2019["DATA LANÇAMENTO"] = pd.to_datetime(df_2019["DATA LANÇAMENTO"], format="%d/%m/%Y")

In [91]:
df_2019.info()

<class 'pandas.core.frame.DataFrame'>
Index: 176828 entries, 63000 to 20000
Data columns (total 15 columns):
 #   Column                     Non-Null Count   Dtype         
---  ------                     --------------   -----         
 0   NOME ÓRGÃO SUPERIOR        176828 non-null  object        
 1   CÓDIGO ÓRGÃO               176828 non-null  int64         
 2   NOME ÓRGÃO                 176828 non-null  object        
 3   CÓDIGO UNIDADE GESTORA     176828 non-null  int64         
 4   NOME UNIDADE GESTORA       176828 non-null  object        
 5   CATEGORIA ECONÔMICA        176828 non-null  object        
 6   ORIGEM RECEITA             176828 non-null  object        
 7   ESPÉCIE RECEITA            176828 non-null  object        
 8   DETALHAMENTO               176828 non-null  object        
 9   VALOR PREVISTO ATUALIZADO  176828 non-null  float64       
 10  VALOR LANÇADO              176828 non-null  float64       
 11  VALOR REALIZADO            176828 non-null  float64   

In [92]:
traducciones = {
    'NOME ÓRGÃO SUPERIOR': 'NOMBRE ÓRGANO SUPERIOR',
    'CÓDIGO ÓRGÃO': 'CÓDIGO ÓRGANO',
    'NOME ÓRGÃO': 'NOMBRE ÓRGANO',
    'CÓDIGO UNIDADE GESTORA': 'CÓDIGO UNIDAD GESTORA',
    'NOME UNIDADE GESTORA': 'NOMBRE UNIDAD GESTORA',
    'CATEGORIA ECONÔMICA': 'CATEGORÍA ECONÓMICA',
    'ORIGEM RECEITA': 'ORIGEN INGRESO',
    'ESPÉCIE RECEITA': 'TIPO INGRESO',
    'DETALHAMENTO': 'DETALLE',
    'VALOR PREVISTO ATUALIZADO': 'VALOR PREVISTO ACTUALIZADO',
    'VALOR LANÇADO': 'VALOR LANZADO',
    'VALOR REALIZADO': 'VALOR REALIZADO',
    'PERCENTUAL REALIZADO': 'PORCENTAJE REALIZADO',
    'DATA LANÇAMENTO': 'FECHA LANZAMIENTO',
    'ANO EXERCÍCIO': 'AÑO EJERCICIO'
}

df_2019 = df_2019.rename(columns=traducciones)

### Nulos

In [93]:
sl.reporte(df_2019)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0              0.00          object
CÓDIGO ÓRGANO                          0              0.00           int64
NOMBRE ÓRGANO                          0              0.00          object
CÓDIGO UNIDAD GESTORA                  0              0.00           int64
NOMBRE UNIDAD GESTORA                  0              0.00          object
CATEGORÍA ECONÓMICA                    0              0.00          object
ORIGEN INGRESO                         0              0.00          object
TIPO INGRESO                           0              0.00          object
DETALLE                                0              0.00          object
VALOR PREVISTO ACTUALIZADO             0              0.00         float64
VALOR LANZADO                          0              0.00         float64
VALOR REALIZADO                        0              0.00         float64
PORCENTAJE REALIZADO                   0              0.00         float64
FECHA LANZAMIENTO                     86              0.05  datetime64[ns]
AÑO EJERCICIO                          0              0.00           int64

In [94]:
df_2019.shape

(176828, 15)

**Manejo de nulos**

En el caso de la base de datos de 2019 vemos que hay un total de 86 nulos en la columna "FECHA DE LANZAMIENTO". Al tener casi 180.000 datos y corresponder a un 0,05% respecto al total de los datos, es factible rellenar los datos nulos con la fecha media para poder cubrir las celdas que no tienen datos, ya que no afectará significativamente al análisis.

In [95]:
df_2019["FECHA LANZAMIENTO"] = df_2019["FECHA LANZAMIENTO"].fillna(df_2019["FECHA LANZAMIENTO"].mean())

In [96]:
sl.reporte(df_2019)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0               0.0          object
CÓDIGO ÓRGANO                          0               0.0           int64
NOMBRE ÓRGANO                          0               0.0          object
CÓDIGO UNIDAD GESTORA                  0               0.0           int64
NOMBRE UNIDAD GESTORA                  0               0.0          object
CATEGORÍA ECONÓMICA                    0               0.0          object
ORIGEN INGRESO                         0               0.0          object
TIPO INGRESO                           0               0.0          object
DETALLE                                0               0.0          object
VALOR PREVISTO ACTUALIZADO             0               0.0         float64
VALOR LANZADO                          0               0.0         float64
VALOR REALIZADO                        0               0.0         float64
PORCENTAJE REALIZADO                   0               0.0         float64
FECHA LANZAMIENTO                      0               0.0  datetime64[ns]
AÑO EJERCICIO                          0               0.0           int64

In [97]:
df_2019.to_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2019_limpio.csv", index=False, encoding='latin-1')

In [98]:
df_2019.sample()

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
26000                  Ministério da Educação          26263   

                                        NOMBRE ÓRGANO  CÓDIGO UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                          
26000                  Universidade Federal de Lavras                 153032   

                                NOMBRE UNIDAD GESTORA CATEGORÍA ECONÓMICA  \
CÓDIGO ÓRGÃO SUPERIOR                                                       
26000                  UNIVERSIDADE FEDERAL DE LAVRAS  Receitas Correntes   

                            ORIGEN INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                        
26000                  Receita de Serviços   

                                                       TIPO INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                                 
26000                  Serviços Administrativos e Comerciais Gerais   

                                                             DETALLE  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
26000                  SERVICOS DE INFORMACAO E TECNOLOGIA-PRINCIPAL   

                       VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                              
26000                                         0.0            0.0   

                       VALOR REALIZADO  PORCENTAJE REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
26000                           5799.3                   0.0   

                      FECHA LANZAMIENTO  AÑO EJERCICIO  
CÓDIGO ÓRGÃO SUPERIOR                                   
26000                        2019-01-02           2019

# 2020

In [99]:
df_2020 = pd.read_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2020.csv",index_col=0, sep=';', encoding='latin-1')

df_2020.head(1)

NOME ÓRGÃO SUPERIOR  CÓDIGO ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                           
63000                  Advocacia-Geral da União         63000   

                                                              NOME ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                  Advocacia-Geral da União - Unidades com víncul...   

                       CÓDIGO UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                           
63000                                  110060   

                                                NOME UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

                      CATEGORIA ECONÔMICA             ORIGEM RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  Receitas Correntes  Outras Receitas Correntes   

                                                   ESPÉCIE RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                                
63000                  Indenizações, restituições e ressarcimentos   

                                                       DETALHAMENTO  \
CÓDIGO ÓRGÃO SUPERIOR                                                 
63000                  RESTITUIÇÃO DE CONVÊNIOS-PRIMÁRIAS-DÍV.ATIVA   

                      VALOR PREVISTO ATUALIZADO VALOR LANÇADO VALOR REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                                           
63000                                      0,00          0,00         5516,90   

                      PERCENTUAL REALIZADO DATA LANÇAMENTO  ANO EXERCÍCIO  
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                                 0,00      28/04/2020           2020

In [100]:
df_2020["VALOR PREVISTO ATUALIZADO"] = df_2020["VALOR PREVISTO ATUALIZADO"].str.replace(',', '.')
df_2020["VALOR PREVISTO ATUALIZADO"] = df_2020["VALOR PREVISTO ATUALIZADO"].astype(float)

In [101]:
df_2020["VALOR LANÇADO"] = df_2020["VALOR LANÇADO"].str.replace(',', '.')
df_2020["VALOR LANÇADO"] = df_2020["VALOR LANÇADO"].astype(float)

In [102]:
df_2020["VALOR REALIZADO"] = df_2020["VALOR REALIZADO"].str.replace(',', '.')
df_2020["VALOR REALIZADO"] = df_2020["VALOR REALIZADO"].astype(float)

In [103]:
df_2020["PERCENTUAL REALIZADO"] = df_2020["PERCENTUAL REALIZADO"].str.replace(',', '.')
df_2020["PERCENTUAL REALIZADO"] = df_2020["PERCENTUAL REALIZADO"].astype(float)

In [104]:
df_2020["DATA LANÇAMENTO"] = pd.to_datetime(df_2020["DATA LANÇAMENTO"], format="%d/%m/%Y")

In [105]:
df_2020.info()

<class 'pandas.core.frame.DataFrame'>
Index: 142348 entries, 63000 to 20000
Data columns (total 15 columns):
 #   Column                     Non-Null Count   Dtype         
---  ------                     --------------   -----         
 0   NOME ÓRGÃO SUPERIOR        142348 non-null  object        
 1   CÓDIGO ÓRGÃO               142348 non-null  int64         
 2   NOME ÓRGÃO                 142348 non-null  object        
 3   CÓDIGO UNIDADE GESTORA     142348 non-null  int64         
 4   NOME UNIDADE GESTORA       142348 non-null  object        
 5   CATEGORIA ECONÔMICA        142348 non-null  object        
 6   ORIGEM RECEITA             142348 non-null  object        
 7   ESPÉCIE RECEITA            142348 non-null  object        
 8   DETALHAMENTO               142348 non-null  object        
 9   VALOR PREVISTO ATUALIZADO  142348 non-null  float64       
 10  VALOR LANÇADO              142348 non-null  float64       
 11  VALOR REALIZADO            142348 non-null  float64   

In [106]:
traducciones = {
    'NOME ÓRGÃO SUPERIOR': 'NOMBRE ÓRGANO SUPERIOR',
    'CÓDIGO ÓRGÃO': 'CÓDIGO ÓRGANO',
    'NOME ÓRGÃO': 'NOMBRE ÓRGANO',
    'CÓDIGO UNIDADE GESTORA': 'CÓDIGO UNIDAD GESTORA',
    'NOME UNIDADE GESTORA': 'NOMBRE UNIDAD GESTORA',
    'CATEGORIA ECONÔMICA': 'CATEGORÍA ECONÓMICA',
    'ORIGEM RECEITA': 'ORIGEN INGRESO',
    'ESPÉCIE RECEITA': 'TIPO INGRESO',
    'DETALHAMENTO': 'DETALLE',
    'VALOR PREVISTO ATUALIZADO': 'VALOR PREVISTO ACTUALIZADO',
    'VALOR LANÇADO': 'VALOR LANZADO',
    'VALOR REALIZADO': 'VALOR REALIZADO',
    'PERCENTUAL REALIZADO': 'PORCENTAJE REALIZADO',
    'DATA LANÇAMENTO': 'FECHA LANZAMIENTO',
    'ANO EXERCÍCIO': 'AÑO EJERCICIO'
}

df_2020 = df_2020.rename(columns=traducciones)

### Nulos

In [107]:
sl.reporte(df_2020)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0              0.00          object
CÓDIGO ÓRGANO                          0              0.00           int64
NOMBRE ÓRGANO                          0              0.00          object
CÓDIGO UNIDAD GESTORA                  0              0.00           int64
NOMBRE UNIDAD GESTORA                  0              0.00          object
CATEGORÍA ECONÓMICA                    0              0.00          object
ORIGEN INGRESO                         0              0.00          object
TIPO INGRESO                           0              0.00          object
DETALLE                                0              0.00          object
VALOR PREVISTO ACTUALIZADO             0              0.00         float64
VALOR LANZADO                          0              0.00         float64
VALOR REALIZADO                        0              0.00         float64
PORCENTAJE REALIZADO                   0              0.00         float64
FECHA LANZAMIENTO                     86              0.06  datetime64[ns]
AÑO EJERCICIO                          0              0.00           int64

In [108]:
df_2020.shape

(142348, 15)

**Manejo de nulos**

En el caso de la base de datos de 2020 vemos que hay un total de 86 nulos en la columna "FECHA DE LANZAMIENTO". Al tener casi 150.000 datos y corresponder a un 0,06% respecto al total de los datos, es factible rellenar los datos nulos con la fecha media para poder cubrir las celdas que no tienen datos, ya que no afectará significativamente al análisis.

In [109]:
df_2020["FECHA LANZAMIENTO"] = df_2020["FECHA LANZAMIENTO"].fillna(df_2020["FECHA LANZAMIENTO"].mean())

In [110]:
sl.reporte(df_2020)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0               0.0          object
CÓDIGO ÓRGANO                          0               0.0           int64
NOMBRE ÓRGANO                          0               0.0          object
CÓDIGO UNIDAD GESTORA                  0               0.0           int64
NOMBRE UNIDAD GESTORA                  0               0.0          object
CATEGORÍA ECONÓMICA                    0               0.0          object
ORIGEN INGRESO                         0               0.0          object
TIPO INGRESO                           0               0.0          object
DETALLE                                0               0.0          object
VALOR PREVISTO ACTUALIZADO             0               0.0         float64
VALOR LANZADO                          0               0.0         float64
VALOR REALIZADO                        0               0.0         float64
PORCENTAJE REALIZADO                   0               0.0         float64
FECHA LANZAMIENTO                      0               0.0  datetime64[ns]
AÑO EJERCICIO                          0               0.0           int64

In [111]:
df_2020.to_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2020_limpio.csv", index=False, encoding='latin-1')

In [112]:
df_2021.sample(2)

NameError: name 'df_2021' is not defined

# 2021

In [86]:
df_2021 = pd.read_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2021.csv",index_col=0, sep=';', encoding='latin-1')

df_2021.head(1)

NOME ÓRGÃO SUPERIOR  CÓDIGO ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                           
63000                  Advocacia-Geral da União         63000   

                                                              NOME ÓRGÃO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                  Advocacia-Geral da União - Unidades com víncul...   

                       CÓDIGO UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                           
63000                                  110060   

                                                NOME UNIDADE GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
63000                  COORD. GERAL DE ORC. FIN. E ANAL. CONT. - AGU   

                       CATEGORIA ECONÔMICA        ORIGEM RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                              
63000                  Receitas de Capital  Operações de Crédito   

                                              ESPÉCIE RECEITA  \
CÓDIGO ÓRGÃO SUPERIOR                                           
63000                  Operações de crédito - mercado interno   

                                                       DETALHAMENTO  \
CÓDIGO ÓRGÃO SUPERIOR                                                 
63000                  TITULOS DE RESPONS.TES.NAC.-MERC.INT.-PRINC.   

                      VALOR PREVISTO ATUALIZADO VALOR LANÇADO VALOR REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                                           
63000                             2224034904,00          0,00            0,00   

                      PERCENTUAL REALIZADO DATA LANÇAMENTO  ANO EXERCÍCIO  
CÓDIGO ÓRGÃO SUPERIOR                                                      
63000                                 0,00      23/04/2021           2021

In [87]:
df_2021["VALOR PREVISTO ATUALIZADO"] = df_2021["VALOR PREVISTO ATUALIZADO"].str.replace(',', '.')
df_2021["VALOR PREVISTO ATUALIZADO"] = df_2021["VALOR PREVISTO ATUALIZADO"].astype(float)

In [88]:
df_2021["VALOR LANÇADO"] = df_2021["VALOR LANÇADO"].str.replace(',', '.')
df_2021["VALOR LANÇADO"] = df_2021["VALOR LANÇADO"].astype(float)

In [89]:
df_2021["VALOR REALIZADO"] = df_2021["VALOR REALIZADO"].str.replace(',', '.')
df_2021["VALOR REALIZADO"] = df_2021["VALOR REALIZADO"].astype(float)

In [90]:
df_2021["PERCENTUAL REALIZADO"] = df_2021["PERCENTUAL REALIZADO"].str.replace(',', '.')
df_2021["PERCENTUAL REALIZADO"] = df_2021["PERCENTUAL REALIZADO"].astype(float)

In [91]:
df_2021["DATA LANÇAMENTO"] = pd.to_datetime(df_2021["DATA LANÇAMENTO"], format="%d/%m/%Y")

In [92]:
df_2021.info()

<class 'pandas.core.frame.DataFrame'>
Index: 134593 entries, 63000 to 20000
Data columns (total 15 columns):
 #   Column                     Non-Null Count   Dtype         
---  ------                     --------------   -----         
 0   NOME ÓRGÃO SUPERIOR        134593 non-null  object        
 1   CÓDIGO ÓRGÃO               134593 non-null  int64         
 2   NOME ÓRGÃO                 134593 non-null  object        
 3   CÓDIGO UNIDADE GESTORA     134593 non-null  int64         
 4   NOME UNIDADE GESTORA       134593 non-null  object        
 5   CATEGORIA ECONÔMICA        134593 non-null  object        
 6   ORIGEM RECEITA             134593 non-null  object        
 7   ESPÉCIE RECEITA            134593 non-null  object        
 8   DETALHAMENTO               134593 non-null  object        
 9   VALOR PREVISTO ATUALIZADO  134593 non-null  float64       
 10  VALOR LANÇADO              134593 non-null  float64       
 11  VALOR REALIZADO            134593 non-null  float64   

In [93]:
traducciones = {
    'NOME ÓRGÃO SUPERIOR': 'NOMBRE ÓRGANO SUPERIOR',
    'CÓDIGO ÓRGÃO': 'CÓDIGO ÓRGANO',
    'NOME ÓRGÃO': 'NOMBRE ÓRGANO',
    'CÓDIGO UNIDADE GESTORA': 'CÓDIGO UNIDAD GESTORA',
    'NOME UNIDADE GESTORA': 'NOMBRE UNIDAD GESTORA',
    'CATEGORIA ECONÔMICA': 'CATEGORÍA ECONÓMICA',
    'ORIGEM RECEITA': 'ORIGEN INGRESO',
    'ESPÉCIE RECEITA': 'TIPO INGRESO',
    'DETALHAMENTO': 'DETALLE',
    'VALOR PREVISTO ATUALIZADO': 'VALOR PREVISTO ACTUALIZADO',
    'VALOR LANÇADO': 'VALOR LANZADO',
    'VALOR REALIZADO': 'VALOR REALIZADO',
    'PERCENTUAL REALIZADO': 'PORCENTAJE REALIZADO',
    'DATA LANÇAMENTO': 'FECHA LANZAMIENTO',
    'ANO EXERCÍCIO': 'AÑO EJERCICIO'
}

df_2021 = df_2021.rename(columns=traducciones)

### Nulos

In [154]:
sl.reporte(df_2021)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0              0.00          object
CÓDIGO ÓRGANO                          0              0.00           int64
NOMBRE ÓRGANO                          0              0.00          object
CÓDIGO UNIDAD GESTORA                  0              0.00           int64
NOMBRE UNIDAD GESTORA                  0              0.00          object
CATEGORÍA ECONÓMICA                    0              0.00          object
ORIGEN INGRESO                         0              0.00          object
TIPO INGRESO                           0              0.00          object
DETALLE                                0              0.00          object
VALOR PREVISTO ACTUALIZADO             0              0.00         float64
VALOR LANZADO                          0              0.00         float64
VALOR REALIZADO                        0              0.00         float64
PORCENTAJE REALIZADO                   0              0.00         float64
FECHA LANZAMIENTO                    123              0.09  datetime64[ns]
AÑO EJERCICIO                          0              0.00           int64

In [155]:
df_2021.shape

(134593, 15)

**Manejo de nulos**

En el caso de la base de datos de 2021 vemos que hay un total de 123 nulos en la columna "FECHA DE LANZAMIENTO". Al tener casi 135.000 datos y corresponder a un 0,09% respecto al total de los datos, es factible rellenar los datos nulos con la fecha media para poder cubrir las celdas que no tienen datos, ya que no afectará significativamente al análisis.

In [156]:
df_2021["FECHA LANZAMIENTO"] = df_2021["FECHA LANZAMIENTO"].fillna(df_2021["FECHA LANZAMIENTO"].mean())

In [157]:
sl.reporte(df_2021)

numero_nulos  porcentaje_nulos  tipo_variables
NOMBRE ÓRGANO SUPERIOR                 0               0.0          object
CÓDIGO ÓRGANO                          0               0.0           int64
NOMBRE ÓRGANO                          0               0.0          object
CÓDIGO UNIDAD GESTORA                  0               0.0           int64
NOMBRE UNIDAD GESTORA                  0               0.0          object
CATEGORÍA ECONÓMICA                    0               0.0          object
ORIGEN INGRESO                         0               0.0          object
TIPO INGRESO                           0               0.0          object
DETALLE                                0               0.0          object
VALOR PREVISTO ACTUALIZADO             0               0.0         float64
VALOR LANZADO                          0               0.0         float64
VALOR REALIZADO                        0               0.0         float64
PORCENTAJE REALIZADO                   0               0.0         float64
FECHA LANZAMIENTO                      0               0.0  datetime64[ns]
AÑO EJERCICIO                          0               0.0           int64

In [158]:
df_2021.to_csv("../11.Proyecto2_EDA-Ingresos-Publicos-Brasil/datos/datos-2021_limpio.csv", index=False, encoding='latin-1')

In [175]:
df_2021.sample(2)

NOMBRE ÓRGANO SUPERIOR  CÓDIGO ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
26000                  Ministério da Educação          26262   
25000                  Ministério da Economia          30203   

                                                           NOMBRE ÓRGANO  \
CÓDIGO ÓRGÃO SUPERIOR                                                      
26000                                  Universidade Federal de São Paulo   
25000                  Instituto Nacional de Metrologia, Qualidade e ...   

                       CÓDIGO UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                          
26000                                 153031   
25000                                 183023   

                                               NOMBRE UNIDAD GESTORA  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
26000                      UNIFESP-UNIVERSIDADE FEDERAL DE SAO PAULO   
25000                  INSTITUTO NAC.DE METROLOG. QUALID. E TECNOLOG   

                      CATEGORÍA ECONÓMICA             ORIGEN INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
26000                  Receitas Correntes        Receita de Serviços   
25000                  Receitas Correntes  Outras Receitas Correntes   

                                                        TIPO INGRESO  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
26000                   Serviços Administrativos e Comerciais Gerais   
25000                  Multas administrativas, contratuais e judicia   

                                                             DETALLE  \
CÓDIGO ÓRGÃO SUPERIOR                                                  
26000                  INSCR.EM CONCURSOS E PROC.SELETIVOS-PRINCIPAL   
25000                  MULTAS PREVISTAS EM LEGISL.ESPECIFICA-DIV.AT.   

                       VALOR PREVISTO ACTUALIZADO  VALOR LANZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                              
26000                                         0.0            0.0   
25000                                         0.0            0.0   

                       VALOR REALIZADO  PORCENTAJE REALIZADO  \
CÓDIGO ÓRGÃO SUPERIOR                                          
26000                          2033.22                   0.0   
25000                         46170.86                   0.0   

                      FECHA LANZAMIENTO  AÑO EJERCICIO  
CÓDIGO ÓRGÃO SUPERIOR                                   
26000                        2021-02-02           2021  
25000                        2021-11-04           2021